lesson19/src/js/news.js

```diff
@@ -0,0 +1,142 @@
+    activeCategory && tabContentsInitialDisplay(categoryButton);
+  });
+  const categoryTabLists = document.querySelector('.tab-list');
+  categoryTabLists.appendChild(fragment);
+}
+
+const getNewsArticle = (articleValue) => {
+  const newsLists = document.createElement('ul');
+  newsLists.classList.add('news-list');
+  const fragment = document.createDocumentFragment();
+  const displayedResults = 5;
+  for (let i = 0; i < displayedResults; ++i){
+    const li = document.createElement('li');
+    li.classList.add('news-item');
+
+    const anchor = document.createElement('a');
+    anchor.classList.add('news-link');
+    anchor.href = articleValue[i].link;
+    anchor.textContent = articleValue[i].title;
+    fragment.appendChild(li).appendChild(anchor);
+    addComment(articleValue[i], li);
+    addNewIcon(articleValue[i],li);
+  }
+  newsLists.appendChild(fragment);
+  return newsLists;
+};
+
+function tabContentsInitialDisplay(target){
+  target.setAttribute('tabindex', '0');
+  target.setAttribute('aria-selected','true');
+
+  const getSelectedTadID = target.getAttribute('aria-controls');
+  const selectedPanel = document.getElementById(getSelectedTadID);
+  selectedPanel.setAttribute('aria-hidden', false);
+}
+
+function addNewsImage(articles){
+  const getNewsArea = document.querySelectorAll('.news-area');
+
+  articles.forEach((category,i) => {
+    const imageArea = document.createElement('div');
+    const img = document.createElement('img');
+    img.src = category.image;
+    img.width = 200;
+    img.height = 200;
+    getNewsArea[i].appendChild(imageArea).appendChild(img);
+  })
+}
+
+function addComment({comment}, item) {
+  const commentResult = comment.length;
+  if (commentResult) {
+    const commentBox = document.createElement('span');
+    const anchor = document.createElement('a');
+    anchor.classList.add('comment-count');
+    anchor.textContent = commentResult;
+    item.appendChild(commentBox).appendChild(anchor);
+  }
+}
+
+function addNewIcon({date},item) {
+  const articleDateToToday = differenceInDays(new Date(),parseISO(date));
+  const judgmenDays = 60;
+  articleDateToToday <= judgmenDays && renderNewIcon(item);
+}
+
+function renderNewIcon(item){
+  const newIcon = createAttributedElements('span',{class:"icon-new"},"new");
+  const newsArticleLink = item.querySelector('.news-link');
+  newsArticleLink.insertAdjacentElement('afterend',newIcon);
+}
+
+export function changeTabs(e) {
+  const tabs = document.querySelectorAll('[role="tab"]');
+  const selectedTab = e.target;
+  tabs.forEach((target) => target.setAttribute('aria-selected', false));
+  selectedTab.setAttribute('aria-selected', true);
+
+  const tabPanels = document.querySelectorAll('[role="tabpanel"]');
+  tabPanels.forEach((tabpanel) => tabpanel.setAttribute('aria-hidden', true));
+  const getSelectedTadID = selectedTab.getAttribute('aria-controls');
+  const selectedPanel = document.getElementById(getSelectedTadID);
```

Comment:
Q
Is this "Tab"??

``const getSelectedTadID``

Reply:
Thank you!
This has been fixed.

[f9334f0](https://github.com/nyattsu72/hanson-frontend-cource/pull/27/commits/f9334f09182730dd2e7fad406dd5989ad86005a8)